src/bytes.rs

```diff
@@ -161,35 +161,12 @@ impl Serialize for Bytes {
     }
 }
 
-struct BytesVisitor;
-
-impl<'de> Visitor<'de> for BytesVisitor {
-    type Value = &'de Bytes;
-
-    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
-        formatter.write_str("a borrowed byte array")
-    }
-
-    fn visit_borrowed_bytes<E>(self, v: &'de [u8]) -> Result<Self::Value, E>
-    where
-        E: Error,
-    {
-        Ok(Bytes::new(v))
-    }
-
-    fn visit_borrowed_str<E>(self, v: &'de str) -> Result<Self::Value, E>
-    where
-        E: Error,
-    {
-        Ok(Bytes::new(v.as_bytes()))
-    }
-}
-
 impl<'a, 'de: 'a> Deserialize<'de> for &'a Bytes {
     fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
     where
         D: Deserializer<'de>,
     {
-        deserializer.deserialize_bytes(BytesVisitor)
+        // serde::Deserialize is already specialized for &[u8], so simply forward to that.
```

Comment:
"Specialized" is misleading to evoke here, because rust's specialization feature is not involved. This comment comes across as though we have an impl for `&[T]` and a specialization of it for `&[u8]` which is not the case. Could you please come up with a different way to describe this?

Reply:
Oh yeah, that makes sense - how about "serde::Deserialize for &[u8] is already optimized"?